streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStoreSupplier.java

```diff
@@ -35,6 +35,8 @@
 /**
  * An in-memory key-value store based on a TreeMap.
  *
+ * Note that array typed keys may result in incorrect behavior for ordering.
```

Comment:
I'd add: "If you are working on byte arrays, for example, you may want to wrap them with the `Bytes` class."